docs/tutorials/tutorial.e2e-ssl.md

```diff
@@ -0,0 +1,200 @@
+          volumeMounts:
+          - mountPath: /etc/nginx/ssl
+            name: secret-volume
+          - mountPath: /etc/nginx/conf.d
+            name: configmap-volume
+      volumes:
+      - name: secret-volume
+        secret:
+          secretName: backend-tls
+      - name: configmap-volume
+        configMap:
+          name: website-nginx-cm
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: website-nginx-cm
+data:
+  default.conf: |-
+    server {
+        listen 8080 default_server;
+        listen 8443 ssl;
+        root /usr/share/nginx/html;
+        index index.html;
+        ssl_certificate /etc/nginx/ssl/tls.crt;
+        ssl_certificate_key /etc/nginx/ssl/tls.key;
+        location / {
+          return 200 "Hello World!";
+        }
+    }
+```
+
+You can also install the above yamls using:
+```bash
+kubectl apply -f https://raw.githubusercontent.com/Azure/application-gateway-kubernetes-ingress/master/docs/examples/sample-https-backend.yaml
+```
+
+Verify that you can curl the application
+```bash
+> kubectl get pods
+NAME                                 READY   STATUS    RESTARTS   AGE
+website-deployment-9c8c6df7f-5bqwh   1/1     Running   0          24s
+website-deployment-9c8c6df7f-wxtnp   1/1     Running   0          24s
+
+> kubectl exec -it website-deployment-9c8c6df7f-5bqwh -- curl -k https://localhost:8443
+Hello World!
+```
+
+## Upload the backend certificate's root certificate to Application Gateway
+When you are setting up SSL between Application Gateway and Backend, if you are using a self-signed certificate or a certificate signed by a custom root CA on the backend, then you need to upload self-signed or the Custom root CA of the backend certificate on the Application Gateway.
+
+```bash
+applicationGatewayName="<gateway-name>"
+resourceGroup="<resource-group>"
+az network application-gateway root-cert create \
+    --gateway-name $applicationGatewayName  \
+    --resource-group $resourceGroup \
+    --name backend-tls \
+    --cert-file backend.crt
+```
+
+## Setup ingress for E2E
+Now, we will configure our ingress to use the `frontend` certificate for frontend SSL and `backend` certificate as root certificate so that Application Gateway can authenticate the backend.
+
+```bash
+cat << EOF | kubectl apply -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: website-ingress
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+    appgw.ingress.kubernetes.io/ssl-redirect: "true"
+    appgw.ingress.kubernetes.io/backend-protocol: "https"
+    appgw.ingress.kubernetes.io/backend-hostname: "backend"
+    appgw.ingress.kubernetes.io/appgw-trusted-root-certificate: "backend-tls"
+spec:
+  tls:
+    - secretName: frontend-tls
+      hosts:
+        - website.com
+  rules:
+    - host: website.com
+      http:
+        paths:
+        - path: /
+          backend:
+            serviceName: website-service
+            servicePort: 8443
```

Comment:
443?

Reply:
this port is the one exposed on the service internally. This port can be anything that user wants to use.